test/Microsoft.AspNet.Mvc.Core.Test/Routing/InnerAttributeRouteTest.cs

```diff
@@ -1464,6 +1464,62 @@ public static IEnumerable<object[]> OptionalParamValues
         }
 
         [Fact]
+        public async Task AttributeRoute_ReplacesExistingRouteValues_IfNotNull()
```

Comment:
Is there a scenario where it can be an empty string? If so we should add that to the condition. But I can't think of a scenario.